### ISSUE TITLE
Download Problems.

### ISSUE THREAD

**Issue by RaketenUwe** [Xoandbit/TreeFarm#3]
I do not understand how to download these files onto the turtle and the computer in Minecraft. I mean downloading it onto my desktop computer (in rl) is quite easy. But I need it ingame. I got all programmes I have used so far from pastebin. There's the "pastebin get" thing. But how does it work with github? Can i do something similar?

**Comment by PPakalns** [comment]
Hi.

There are two solutions:

**First option:**
If your server has enabled Pastebin then probably HTTP computercraft API will work.
Then you can write the following computercraft script to download raw files directly from the GitHub repository.
```lua
function download(url, file)
  local content = http.get(url).readAll()
  if not content then
    error("Could not connect to website")
  end
  f = fs.open(file, "w")
  f.write(content)
  f.close()
end

-- Download TreeFarm programs
download("https://raw.githubusercontent.com/PPakalns/TreeFarm/master/control.lua", "control.lua")
download("https://raw.githubusercontent.com/PPakalns/TreeFarm/master/lain.lua", "lain.lua")
download("https://raw.githubusercontent.com/PPakalns/TreeFarm/master/worker.lua", "worker.lua")
```
Source for download function:
http://www.computercraft.info/forums2/index.php?/topic/11809-lua-downloading-files-with-http/page__view__findpost__p__105952

**Second option:**
Copy the raw files to Pastebin and download them from there using Pastebin command.

---

Hope that this will help and will resolve the question.

If you have any other difficulties setting up TreeFarm please feel free to ask any questions. (The instructions sadly aren't in the best state :) )

**Comment by RaketenUwe** [comment]
so after trying it a lot more times I am now stuck with the turtle spamming "SENDING STARTUP" and nothing else happens. It seems that although they are connected it doesnt really send it.

**Comment by PPakalns** [comment]
It means that it cannot connect with the server computer.

Please make sure that the server computer is running correctly and is executing file that contains control.lua content. For debug purposes you can attach a monitor to the back of the server computer to see some debug output.

It should at least output the top line with the name "Tree Farm" and a number which will increase.

It shows connected turtles and list of jobs in a cryptic way.

![Example setup](https://raw.githubusercontent.com/PPakalns/TreeFarm/development/screenshots/y.png)



_I haven't tested if this farm works on the newest computercraft version because there could be some API changes. But as the program hasn't crashed, it probably will work._

**Comment by RaketenUwe** [comment]
I havent got these output. The Computer itself says "starting tree farm" and the monitor "- Tree farm - time 0 x 2"

**Comment by PPakalns** [comment]
Haven't played Minecraft for a long time and this seems like a good reason to pick up it again.
Which Minecraft and computercraft version do you have? (Do you use a specific modpack from the technic launcher?)
Will try to setup it myself and update instructions if necessary.

**Comment by RaketenUwe** [comment]
thank you for your effort ^^. We are playing the FTB Infinity Evolved modpack from the FTB/Twitch launcher.

**Comment by PPakalns** [comment]
Hi, I got it working on Switchcraft mod pack.

Some things to keep in mind, please try at least the first setup of TreeFarm **without any advanced computer parts.** I haven't tested advanced computer support with this program (and they are not necessary for it)

Second, facing coordinate is not displayed on newer Minecraft versions:
It can be deduced by using the following quote:
> Among the other information in the debug screen, the item "f" gives your facing, meaning which compass direction you are (most nearly) facing. A value of 0 means south, 1 west, 2 north, and 3 east. 

But everything else worked when following my cryptic instructions.

Can you check that your computer and turtle has modem placed on the correct side? (On the right side like in the images)
It seems like in your case they cannot connect with each other (check that they both use normal (not advanced) modems and that they have modems attached to the correct side)

**Comment by RaketenUwe** [comment]
Hi
So, the modems should be placed correct, both on the right side. And in this right and right  setup of turtle and computer was the first time I made more progress than before. I think it it important to mention that I am playing in the 1.7.10 Version. This means that the face coordinate is still visible.
And we thought we made 2 correct tries, exaclty like you described it, but it didnt work 2 times, having the "error" I described before. By that I mean the "sending startup". 
I havent tried the modem on the left side of the computer yet, might aswell be trying that later.

**Comment by PPakalns** [comment]
No need to try the left side.
The modem side is configured at the start of worker.lua, control.lua files in `modemSide` variable.

I downloaded FTB Infinity Evolved through FTB launcher and tested TreeFarm in single player. It worked.

Can you check that modem on turtle and control computer turns red when programs are running.

At least now I don't have any other suggestion, can you, please, provide more additional information about your setup, maybe some images, are you trying to use it in multiplayer environment or singleplayer, etc.

**Comment by RaketenUwe** [comment]
Okay. In the hope that the "Comment" button will work this time. 

I sadly dont have any time to send pictures today. But I can tell that both Modems were lighted red once the programmes were running. We are playing on a dedicated Server of mine, but a friend has also tried it in Singleplayer and it didnt work for him either . So it seems we are still doing something wrong, when your Singleplayer works, but ours doesnt. 

Have a nice Christmas Eve
Greetings :)

**Comment by PPakalns** [comment]
Hi, if you want, you can try adding me on discord PPakalns#2579 and if we are online and free at the same time we can try to setup TreeFarm in public (creative?) computercraft server.

Thanks, Merry Christmas to you, too!

**Comment by RaketenUwe** [comment]
I sent you a friend request :)

**Comment by RaketenUwe** [comment]
My name is Tilted if you wonder what it is ^^

**Comment by PPakalns** [comment]
Fixed the issue: afb65c998fea903b1ff3ad5530e32727e2bf178c
